lib/em-mongo/cursor.rb

```diff
@@ -89,6 +89,7 @@ def next_document
         refresh.callback do
           check_and_transform_document(@cache.shift, response)
         end
+        refresh.errback { |err| response.fail err }
```

Comment:
This line calls refresh a second time, requesting more data from the db.
Also errback is then set on a different Deferrable than the one with callback, possibly also causing it to never get called.

If this isn't intended, please use a object similar to line 428.

Reply:
thx!
I totally missed that - I'm not doing that much ruby coding and missed the fact that refresh is actually a function invocation and not field/variable

I'll fix that of course